### PR TITLE
Add poster and order lists

### PR DESCRIPTION
**Pull request by slax57**
- [x] Enable TS in repo (was still in JS mode somehow)
- [x] Add `vite` config
- [x] Add fake rest data provider, hooked to [data-generator-retail](https://github.com/marmelab/react-admin/tree/master/examples/data-generator)
- [x] Add simple _Posters_ list
- [x] Add simple _Orders_ list

### CODE REVIEW

File: src/command/CommandList.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      <TextField source="reference" />
+      <DateField source="date" />
+      <ReferenceField source="customer_id" reference="customers">
+        <TextField source="id" />
+      </ReferenceField>
+      <ArrayField source="basket">
+        <SingleFieldList>
+          <ChipField source="product_id" />
+        </SingleFieldList>
+      </ArrayField>
```
Comment (djhi) [inline src/command/CommandList.tsx:26]: You should avoid this kind of fields in list. There might be many products in the basket

File: src/product/ProductList.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <Datagrid rowClick="edit">
+      <TextField source="id" />
+      <ReferenceField source="category_id" reference="categories">
+        <TextField source="id" />
+      </ReferenceField>
+      <NumberField source="width" />
+      <NumberField source="height" />
+      <NumberField source="price" />
+      <TextField source="thumbnail" />
+      <TextField source="image" />
```
Comment (djhi) [inline src/product/ProductList.tsx:20]: ImageField maybe?

**Review by djhi — CHANGES_REQUESTED** [review]

File: src/command/CommandList.tsx (hunk 1 above)
Comment (slax57) [reply src/command/CommandList.tsx:26]: Fixed in https://github.com/slax57/admin-e-commerce-demo/pull/2 🙂

File: src/product/ProductList.tsx (hunk 2 above)
Comment (slax57) [reply src/product/ProductList.tsx:20]: Fixed in https://github.com/slax57/admin-e-commerce-demo/pull/2 🙂

**Pull request merged**